src/entities/post/models/queries.ts

```diff
@@ -0,0 +1,97 @@
+import { useQuery, useMutation, useQueryClient } from "@tanstack/react-query";
+import { postsKeys } from "../../../shared/query-keys/post";
+import { getPosts, getPostBySearch, getPostByTag, addPost, updatePost, deletePost } from "./api";
+import type { PageParams } from "../../../shared/query-keys/post";
+import type { Post, PostsResponse } from "./types";
+
+export function usePostsPageQuery(params: PageParams) {
+  return useQuery<PostsResponse>({
+    queryKey: postsKeys.page(params),
+    queryFn: () => getPosts(params),
+    placeholderData: (prev) => prev,
+    staleTime: 30_000,
+    gcTime: 5 * 60_000,
+    refetchOnWindowFocus: false,
+    retry: 1,
+  });
+}
+
+export function usePostsSearchQuery(search: string, enabled = true) {
+  return useQuery<PostsResponse>({
+    queryKey: postsKeys.search(search),
+    queryFn: () => getPostBySearch({ search }),
+    enabled: enabled && !!search,
+    staleTime: 30_000,
+    gcTime: 5 * 60_000,
+  });
+}
+
+export function usePostsByTagQuery(tag: string, enabled = true) {
+  return useQuery<PostsResponse>({
+    queryKey: postsKeys.tag(tag),
+    queryFn: () => getPostByTag({ tag }),
+    enabled: enabled && !!tag && tag !== "all",
+    staleTime: 30_000,
+    gcTime: 5 * 60_000,
+  });
+}
+
+function updateAllPostLists(qc: ReturnType<typeof useQueryClient>, updater: (prev: PostsResponse) => PostsResponse) {
+  const lists = qc.getQueriesData<PostsResponse>({ queryKey: postsKeys.all });
+  lists.forEach(([key, prev]) => {
+    if (!prev) return;
+    qc.setQueryData<PostsResponse>(key, updater(prev));
+  });
+}
+
+export function useAddPostMutation() {
+  const qc = useQueryClient();
+  return useMutation({
+    mutationFn: ({ post }: { post: Partial<Post> }) => addPost({ post }),
+    onSuccess: (created) => {
```

Comment:
뭔가 이런 식으로 타이핑이 안되어서 밑에서 as 를 쓰셨나요?
addPost 가 Promise<Post> 를 잘 반환하고 있어서 왠지 Post 를 잘 내보내고 있어야할 것 같았어요.

```suggestion
    onSuccess: (created: Post) => {
```